experiments/replicaset.go

```diff
@@ -199,6 +204,56 @@ func (ec *experimentContext) isReplicaSetSemanticallyEqual(newRS, existingRS *ap
 		existingAnnotations[v1alpha1.ExperimentTemplateNameAnnotationKey] == newAnnotations[v1alpha1.ExperimentTemplateNameAnnotationKey]
 }
 
+// addScaleDownDelay injects the `scale-down-deadline` annotation to the ReplicaSet, or if
+// scaleDownDelaySeconds is zero, removes it if it exists
+// returns True if ReplicaSet is patched, otherwise False
+func (ec *experimentContext) addScaleDownDelay(rs *appsv1.ReplicaSet) (bool, error) {
+	rsIsUpdated := false
+	if rs == nil {
+		return rsIsUpdated, nil
+	}
+	ctx := context.TODO()
+	scaleDownDelaySeconds := time.Duration(defaults.GetExperimentScaleDownDelaySecondsOrDefault(ec.ex))
+	if scaleDownDelaySeconds == 0 {
+		// If scaledown deadline is zero, it means we need to remove any replicasets with the delay
+		if replicasetutil.HasScaleDownDeadline(rs) {
+			return ec.removeScaleDownDelay(rs)
+		}
+		return rsIsUpdated, nil
+	} else {
+		// If RS already has non-zero scaleDownDelayDeadline set, then we don't do anything
+		if replicasetutil.HasScaleDownDeadline(rs) {
+			return rsIsUpdated, nil
+		}
+	}
+
+	deadline := metav1.Now().Add(scaleDownDelaySeconds * time.Second).UTC().Format(time.RFC3339)
```

Comment:
scaleDownDelaySeconds * time.Second ..what this mean? is it not addition instead of multiplying?

Reply:
`deadline := metav1.Now().Add(scaleDownDelaySeconds * time.Second).UTC().Format(time.RFC3339)`

This line creating a deadline is adding `scaleDownDelaySeconds` to the current time.